Hyperion/SerializerFactories/ExceptionSerializerFactory.cs

```diff
@@ -37,19 +37,31 @@ public ExceptionSerializerFactory()
 
         public override bool CanDeserialize(Serializer serializer, Type type) => CanSerialize(serializer, type);
 
+        // Workaround for CoreCLR where FormatterServices.GetUninitializedObject is not public
+        private static readonly Func<Type, object> GetUninitializedObject =
+            (Func<Type, object>)
+                typeof(string).GetTypeInfo().Assembly.GetType("System.Runtime.Serialization.FormatterServices")
+                    .GetMethod("GetUninitializedObject", BindingFlags.NonPublic | BindingFlags.Public | BindingFlags.Static)
+                    .CreateDelegate(typeof(Func<Type, object>));
+
+        private static object FormatterServicesGetUninitializedObject(Type type) => GetUninitializedObject.Invoke(type);
+
         public override ValueSerializer BuildSerializer(Serializer serializer, Type type,
             ConcurrentDictionary<Type, ValueSerializer> typeMapping)
         {
             var exceptionSerializer = new ObjectSerializer(type);
+            var hasDefaultConstructor = type.GetTypeInfo().GetConstructor(new Type[0]) != null;
             exceptionSerializer.Initialize((stream, session) =>
             {
-                var exception = Activator.CreateInstance(type);
                 var className = stream.ReadString(session);
                 var message = stream.ReadString(session);
                 var remoteStackTraceString = stream.ReadString(session);
                 var stackTraceString = stream.ReadString(session);
                 var innerException = stream.ReadObject(session);
 
+                var exception = hasDefaultConstructor ? Activator.CreateInstance(type) :
+                    FormatterServicesGetUninitializedObject(type);
```

Comment:
You don't need this check here - it's not like having default constructor or not will change depending on the binary payload ;) You can move this check outside the deserializer function i.e: 

```csharp
Func<Object> initialize = hasDefaultConstructor 
    ? (() => Activator.CreateInstance(type)) 
    : (FormatterServicesGetUninitializedObject(type));
```
and use `initialize` inside deserializer.

Reply:
But now it introduces unnecessary delegate allocation.

Reply:
It happens once per exception type, while constructor check will happen once per each object deserialization call. In practice to be 100% sure, we'd need to check both approaches in performance test - I'll be happy to do that once #57 will finally get merged.

Reply:
You're right, thanks for the suggestion.